### PR TITLE
fix(frontend): Bug in pushdown of `now()`

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
Sorry, noticed mistakes in https://github.com/risingwavelabs/risingwave/pull/6964 but it got merged before I fixed it.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
https://github.com/risingwavelabs/risingwave/pull/6964

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/logical_filter.rs [hunk 1]
```diff
@@ -236,7 +236,11 @@ impl ToStream for LogicalFilter {
...
                                 != Type::Now
-                                || !matches!(&now_expr.inputs()[1], ExprImpl::Literal(_))
+                                || !matches!(
+                                    &now_expr.inputs()[1],
+                                    ExprImpl::Literal(_) | ExprImpl::FunctionCall(_)
+                                )
+                                || now_expr.inputs()[1].has_input_ref()
                             {
                                 return Err(ExprError::InvalidParam {
                                     name: "now",
```
Comment (soundOfDestiny) [inline src/frontend/src/optimizer/plan_node/logical_filter.rs:246]: error msg here should be adjusted accordingly

**Review by soundOfDestiny — APPROVED** [review]
rest lgtm

**Comment by soundOfDestiny** [comment]
thanks

File: src/frontend/src/optimizer/plan_node/logical_filter.rs (hunk 1 above)
Comment (jon-chuang) [reply src/frontend/src/optimizer/plan_node/logical_filter.rs:246]: Hmm. Can we say "delta expression cannot reference any columns"?

Comment (jon-chuang) [reply src/frontend/src/optimizer/plan_node/logical_filter.rs:246]: Actually, I don't want to confuse the user. I think the general use-case is to use a literal like '1 hr'. So I suggest we stick with this msg.

**Pull request merged**